mailpoet/lib/Subscribers/SubscriberSaveController.php

```diff
@@ -277,4 +294,27 @@ public function updateCustomFields(array $data, SubscriberEntity $subscriber): v
       $this->subscriberCustomFieldRepository->createOrUpdate($subscriber, $customField, $customFieldsMap[$customField->getId()]);
     }
   }
+
+  private function updateTags(array $data, SubscriberEntity $subscriber): void {
+    foreach ($subscriber->getSubscriberTags() as $subscriberTag) {
+      $tag = $subscriberTag->getTag();
+      if (!$tag) {
+        continue;
```

Comment:
>If I'm not mistaken this happens in case the tag entity was deleted but the association is still saved in DB. Am I right?

I think soo.

>If so. I think we could cleanup the association by calling $subscriber->getSubscriberTags()->removeElement($subscriberTag); instead of continue.

I agree with your proposal. I will change it.